### PR TITLE
Update kwargs validation for `preprocess` with decorator

### PR DESCRIPTION
**Pull request by qubvel**
# What does this PR do?

Update most of the image processors with `filter_out_non_signature_kwargs` decorator for kwargs validation instead of `_valid_processor_keys` (continue of work started in https://github.com/huggingface/transformers/pull/30799)

## Before submitting
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qubvel** [comment]
CLIP/Git image processors require updating their processors first, DETR-like model requires adding deprecation. In this PR **only** simple cases are handled, where we need to replace `_valid_processor_keys` with a decorator, other cases will be done in follow-up PRs for review simplicity.

**Comment by qubvel** [comment]
@molbap whenever you have time, please have a look

**Review by molbap — APPROVED** [review]
LGTM! Much tidier. One comment is since now `inspect.signature` is used everywhere, is there any impact on tests speed?

**Comment by qubvel** [comment]
@molbap `inspect.signature` is used only once the class method when the class is defined (on class import), so I would not expect any slowdown. Even if we have 1k imports of objects overall it should not take longer than a fraction of a second. Am I missing something here? Afair, you choose to use `_valid_processor_kwargs` instead `inspect.signature`, was there any specific reason?

**Comment by molbap** [comment]
Yes you're right, it's only at import time. I was just curious since inspect-related calls have ended up costing me some precious ms before, so I steer away from it

**Comment by qubvel** [comment]
@amyeroberts please review whenever you have time

File: tests/test_image_processing_common.py [hunk 1]
```diff
@@ -425,15 +426,41 @@ def test_call_numpy_4_channels(self):
...
+                with warnings.catch_warnings(record=True) as raised_warnings:
+                    warnings.simplefilter("always")
+                    image_processor(inputs, extra_argument=True)
+
+                messages = " ".join([str(w.message) for w in raised_warnings])
+                self.assertGreaterEqual(len(raised_warnings), 1)
+                self.assertIn("extra_argument", messages)
+                is_tested = True
+
+        if not is_tested:
```
Comment (amyeroberts) [inline tests/test_image_processing_common.py:462]: Nice :)

**Review by amyeroberts — APPROVED** [review]
Beautiful ❤️

**Pull request merged**